sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/MessageBody.cs

```diff
@@ -162,5 +170,82 @@ private void Append(Data segment)
                 _segments.Add(memory.Slice(0, dataToAppend.Length));
             }
         }
+
+        /// <summary>
+        /// Eagerly copies the provided data segments into a single continuous buffer while still keeping around a list of the individual copied segments.
+        /// Important for the receive path in order to make sure the buffers managed by the underlying AMQP library can be released on dispose.
+        /// </summary>
+        private sealed class EagerCopyingMessageBodyWithManagedBuffer : MessageBody
```

Comment:
Can this be combined with the existing EagerCopyingMessageBody?

Reply:
The only thing that is duplicated is 

https://github.com/Azure/azure-sdk-for-net/pull/20320/files/57bb61caf5b8538d275a38ed7942a25515cfc8d4#diff-fff97bdd6234f68d4f57b5227fefa307226ba6aae7f8df9f1f8726368697d6a3R228-R236

the rest is highly specific to the optimized structure. Honestly I try to not over abstract until things have really enough similarity. Especially not in performance sensitive areas because the baggage of bringing around fields and other stuff that might not be used can quickly become a burden. I'm happy to move though the mapping method into some common place in that file. What do you think?

Reply:
Yeah, extracting the mapping method would be great.

Reply:
Done